ykcompile/src/lib.rs

```diff
@@ -222,6 +234,63 @@ impl TraceCompiler {
         Ok(())
     }
 
+    /// Compile a call to a native symbol. This is used for occasions where you don't want the
+    /// callee inlined, or cannot inline it (e.g. it's a foreign function).
+    fn c_call(
+        &mut self,
+        opnd: &CallOperand,
+        args: &Vec<Operand>,
+        dest: &Option<Place>,
+    ) -> Result<(), CompileError> {
+        let sym = if let CallOperand::Fn(sym) = opnd {
+            sym
+        } else {
+            return Err(CompileError::Unimplemented(
+                "unknown call target".to_owned(),
+            ));
+        };
+
+        // FIXME implement support for arguments.
+        if !args.is_empty() {
+            return Err(CompileError::Unimplemented("call with args".to_owned()));
+        }
+
+        // Generally speaking, we don't know what the callee will clobber, so we save our registers
```

Comment:
s/clobber/overwrite?

Reply:
"Clobber" is the normal term here (for better or worse), in the sense that it implies an element of chaos to what the callee might do.

Reply:
Ah, I did not know that. Not the word I would have used.

Reply:
It's a GNU assembler term, yeah:

![image](https://user-images.githubusercontent.com/604955/82880407-e99dd080-9f2d-11ea-92d8-e26d40327b8d.png)


https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html

Reply:
It's much more widely used than just GNU's assembler!

Reply:
Fair enough! I don't use many other assemblers these days!